### ISSUE TITLE
TypeScript complains about overwriting .d.ts files

### ISSUE THREAD

**Issue by sguillia** [microsoft/TypeScript#35541]
**TypeScript Version:**  Version 3.8.0-dev.20191206

**Search Terms:** overwrite .d.ts declaration

**Steps to reproduce**

The repo is a minimal working example and contains practically no code.

```
git clone https://github.com/sguillia/ts-bug-repro
cd ts-bug-repro
tsc -p . # Ignore error about missing .d.ts
tsc -p .
```

**Expected behavior:**

Compiles without error, by ignoring the `dist` folder as specified in the `tsconfig.json`.

**Actual behavior:**

The second `tsc` run spits out an error.
```
error TS5055: Cannot write file '/home/user/ts-bug-repro/dist/main.d.ts' because it would overwrite input file.

Found 1 error.
```

**Related Issues:**

Not a valid duplicate because it is old, misses version information and misses easy-to-reproduce steps
https://github.com/Microsoft/TypeScript/issues/16749

Look for `error TS5055`. The FAQ mentions this error but takes the assumption that we have `allowJs` which is untrue in my case.
https://github.com/Microsoft/TypeScript/wiki/FAQ#common-bugs-that-arent-bugs

**Comment by RyanCavanaugh** [comment]
I'm not sure what to say; the error message is accurate and you've configured your project in a way to produce the problem described by the error message. TypeScript will not proceed with a compilation where an output overwrites an input; you can either fix this by having inputs in a different location or outputs in a different location, it's up to you.

**Comment by sguillia** [comment]
Dear @RyanCavanaugh ,

I would like all my inputs to be in `src`.
I would like all my outputs to be in `dist`.

This means that I would like typescript to consider the `.d.ts` files as output files since they are in `dist`.

I did not manage to do this.
I am confused as I tried to set `rootDir`, `outDir`, `include` and `exclude` accordlingly [here](https://github.com/sguillia/ts-bug-repro/blob/master/tsconfig.json) but I still have the error.

How can I get Typescript to not consider the `.d.ts` as input files?

Many thanks
